File: contest/abc326/a.rb
x, y = gets.chomp.split(" ").map(&:to_i)

if x > y
  if x - 3 <= y
    puts "Yes"
  else
    puts "No"
  end
else
  if x + 2 >= y
    puts "Yes"
  else
    puts "No"
  end
end

File: contest/abc326/a.exs
defmodule Main do
  def main do
    IO.read(:line)
    |> String.trim
    |> String.split(" ")
    |> Enum.map(&String.to_integer/1)
    |> solve
  end

  def solve([x, y]) do
    if x < y do
      if x + 2 >= y do
        "Yes"
      else
        "No"
      end
    else
      if x - 3 <= y do
        "Yes"
      else
        "No"
      end
    end
    |> IO.puts
  end
end

Main.main()